src/ucp/wireup/address.c

```diff
@@ -279,6 +279,27 @@ ucp_address_unpack_worker_address_name(const void *src, char *s)
     return UCS_PTR_TYPE_OFFSET(UCS_PTR_BYTE_OFFSET(src, length), uint8_t);
 }
 
+static uint8_t ucp_address_pack_rel_version()
+{
+    return UCP_API_MINOR % UCS_BIT(UCP_ADDRESS_REL_VERSION_BITS);
+}
+
+/* Release version is packed into 4 bits. Unpack it assuming that we do not
+ * support wire compatibility for more than 8 versions (therefore we take the
+ * value which is closest to our release version)
+ */
+static unsigned ucp_address_unpack_rel_version(uint8_t packed_version)
+{
+    int range = UCS_BIT(UCP_ADDRESS_REL_VERSION_BITS);
+    int base, delta;
+
+    delta = ucp_address_pack_rel_version() - packed_version;
+    base  = UCP_API_MINOR / range + (delta >= range / 2) -
+                                               (delta <= -range / 2);
+
+    return packed_version + ucs_max(0, base) * range;
```

Comment:
can we simplify?

Reply:
1. define UCS_BIT(UCP_ADDRESS_REL_VERSION_BITS) as macro and reuse in ucp_address_pack_rel_version
2. init delta inline
3. add ( ) around internal expressions